File: hw8/facts.py
ans = 1

print("n\tn!\n-----------")

def resetans():
    ans = 1

def factorial(n):
    resetans()
    if n < 1:
        return ans
    else:
        return n * factorial(n-1)

n = 0
while n < 6:
    print(n, end='\t')
    print(factorial(n))
    print()
    n = n + 1

File: hw8/Facts.java
class Facts
{
    static long factorial(int par)
    {
	long answer = 1;
	if (par == 0)
	    return answer;
	int c  = 1;
	while (c <= par)
	    {
		answer = answer * c;
		c++;
	    }
	return answer;
    }

    public static void main(String args[])
    {
	System.out.println("n" + "       " + "n!");
	System.out.println("------------");
	int n = 0;
	while (n <= 6)
	    {
		System.out.println(n + "       " + factorial(n));
		n++;
	    }
	return;
    }
}
